File: study/lua/src/app/ui/PopupSkillDetail.lua
-- Author: hedili
-- Date: 2018-01-24 15:09:42
-- 技能详情
local PopupBase = require("app.ui.PopupBase")
local PopupSkillDetail = class("PopupSkillDetail", PopupBase)
local TypeConvertHelper = require("app.utils.TypeConvertHelper")
local PopupSkillDetailCell = require("app.ui.PopupSkillDetailCell")

function PopupSkillDetail:ctor(skillDescList)
	self._skillDescList = skillDescList or {}
	self._listView = nil
	local resource = {
		file = Path.getCSB("PopupSkillDetail", "common"),
		binding = {

		}
	}

	PopupSkillDetail.super.ctor(self, resource)
end

function PopupSkillDetail:onCreate()
	self._listView:removeAllChildren()
	self._commonBK:setTitle(Lang.get("common_skill_detail"))
	self._commonBK:addCloseEventListener(handler(self, self.close))
	
	for i, skillValue in ipairs(self._skillDescList) do
		local skillCell = PopupSkillDetailCell.new()
		skillCell:updateUI(skillValue.title, skillValue.skillId,skillValue.pendingStr)
		self._listView:pushBackCustomItem(skillCell)
	end
end

function PopupSkillDetail:onEnter()

end

function PopupSkillDetail:onExit()

end


return PopupSkillDetail

File: study/ts/script/ui/PopupSkillDetail.ts
const { ccclass, property } = cc._decorator;

import CommonNormalMidPop from './component/CommonNormalMidPop'
import CommonCustomListView from './component/CommonCustomListView';
import PopupBase from './PopupBase';
import { Lang } from '../lang/Lang';
import { handler } from '../utils/handler';
import PopupSkillDetailCell from './PopupSkillDetailCell';

@ccclass
export default class PopupSkillDetail extends PopupBase {
    public static path = 'common/PopupSkillDetail';

    @property({
        type: CommonNormalMidPop,
        visible: true
    })
    _commonBK: CommonNormalMidPop = null;

    @property({
        type: CommonCustomListView,
        visible: true
    })
    _listView: CommonCustomListView = null;

    @property(cc.Prefab)
    popupSkillDetailCell: cc.Prefab = null;
    _skillDescList: any;

    ctor(skillDescList) {
        this._skillDescList = skillDescList || {};
    }
    onCreate() {
        this._listView.removeAllChildren();
        this._commonBK.setTitle(Lang.get('common_skill_detail'));
        this._commonBK.addCloseEventListener(handler(this, this.close));
        for (var i in this._skillDescList) {
            var skillValue = this._skillDescList[i];
            var skillCell = cc.instantiate(this.popupSkillDetailCell).getComponent(PopupSkillDetailCell);
            skillCell.updateUI(skillValue.title, skillValue.skillId, skillValue.pendingStr);
            this._listView.pushBackCustomItem(skillCell.node);
        }
    }
    onEnter() {
    }
    onExit() {
    }
}